code/game/objects/items/oddities.dm

```diff
@@ -32,6 +32,7 @@
 		for(var/stat in oddity_stats)
 			oddity_stats[stat] = rand(1, oddity_stats[stat])
 		AddComponent(/datum/component/inspiration, oddity_stats)
+		AddComponent(/datum/component/artifact_power, oddity_stats)
```

Comment:
everything about this new component is wrong. What you should do use the inspiration component and adapt it to your needs (adding new variables and procs).

Reply:
I think I did as you asked